### ISSUE TITLE
review opportunities for speeding up some of the models run during sessions

### ISSUE THREAD

**Issue by sbfnk** [nfidd/nfidd#220]
With the longer data set the random walk models are very slow (>5 mins on a modern macbook pro, could easily be >30 minutes on a standard laptop).

Options are:
- save data sets that then get included in the package (not sure how we'd do it out of the `qmd` given we can't use `echo = FALSE` with visual editor)
- revert to the smaller data set (though makes the nowcasts look less good)
- others?

**Comment by seabbs** [comment]
> With the longer data set the random

by longer data do you mean a cutoff of 71 or higher? If its at 71 I think we need to take an action if its some custom setting you are getting from exploring it seems not ideal but fine for this time? 

> others?

Is making it weekly I think + making sure we are actually setting parallel cores everywhere.

**Comment by sbfnk** [comment]
> by longer data do you mean a cutoff of 71 or higher? If its at 71 I think we need to take an action if its some custom setting you are getting from exploring it seems not ideal but fine for this time?

I mean that we now have an outbreak lasting 140 days thanks to larger pop size

**Comment by sbfnk** [comment]
> From this it sounds like it is in the current sessions? Not sure it can be as the action only takes 28 minutes

hmm true the actions are still quickish so perhaps it's ok

**Comment by seabbs** [comment]
> I mean that we now have an outbreak lasting 140 days thanks to larger pop size 

got you. If you fit too 100+ data points I see really long run times. We could put a warning in somewhere?

**Comment by sbfnk** [comment]
Yes, and maybe review later whether we really can't get a way with a smaller outbreak (or alternatively: one with a higher R0 as it'll be over quicker).

**Comment by seabbs** [comment]
Action that is pre course is add a warning about long run times with a high number of data points. 

Action post course is to look at potential speed ups that dobt sacrifice clarity

**Comment by seabbs** [comment]
My vote is not do this time

**Comment by seabbs** [comment]
We could try tightening the variance for the random walk models?
